### PR TITLE
Add automated workflow publishing capabilities

### LINKED ISSUE #18: Add automated GitHub publishing capabilities
Add automated GitHub publishing capabilities which includes HTML, PDF, and ePub content.

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This PR adds the capability to publish the Almanac's content automatically through GitHub Actions to address #18 . This adds HTML and PDF content through GitHub Pages. I tried to include ePub with this work but found that it's more complicated than the HTML or PDF content publishing process. As a result, I plan to address that aspect outside this PR, but I don't consider it critical (as both an offline and online option are provided here).

I also noticed that:
- the logo wasn't working when linked from an HTTP reference, so updated this.
- publishing wasn't looking correct with a preface placeholder so I removed it entirely.

Thanks for any feedback you may have!

Closes #18 (#30 created for ePub tracking as follow-up)

## What is the nature of your change?

- [ ] Content additions or updates (adds or updates content)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own contributions.
- [x] I have commented my content, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to related documentation (outside of book content).
- [x] My changes generate no new warnings.
- [x] New and existing tests pass locally with my changes.
- [ ] I have added tests that prove my additions are effective or that my feature works.
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

**Review by falquaddoomi — APPROVED** [review]
Neat! All the changes look pretty straightforward. I also tried out the `build-book` and `build-book-pdf` commands locally, both of which ran to completion (including fetching Chromium, which was cool to see) and produced nice-looking output.

I've noticed that, in the PDF version of the book and when you select the ".pdf" option from the "Download" dropdown in the HTML version, the PDF has a "Skip to main content" overlay on each page that covers up some of the content. Here's an example from the PDF:

Screenshot 2024-05-09 at 9 24 20 AM

I'm approving this PR since that issue is out of scope for automating the build process, but I thought I'd mention it. I can file an issue for this one and these sorts of things in the future, if you like.

**Review by gwaybio — APPROVED** [review]

**Comment by d33bs** [comment]
Thanks @falquaddoomi and @gwaybio for the reviews!

> I've noticed that, in the PDF version of the book and when you select the ".pdf" option from the "Download" dropdown in the HTML version, the PDF has a "Skip to main content" overlay on each page that covers up some of the content.

Thanks for bringing this up! I did some investigation and found that this relates to https://github.com/executablebooks/jupyter-book/issues/2045 and https://github.com/pydata/pydata-sphinx-theme/issues/1745 . I've added a bit of CSS to avoid displaying the "skip" content when printing through Chromium (which is how the PDF is compiled).

File: README.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 An open-source handbook of applied guidance and tools for sustainable software development and maintenance.
 
+Please see the following links to access published content:
+
+- __Online (HTML)__: https://software-gardening.github.io/almanac/
+- __Offline (PDF)__: [software-gardening-almanac.pdf](https://software-gardening.github.io/almanac/software-gardening-almanac.pdf)
```
Comment (vincerubinetti) [inline README.md:8]: Sorry I didn't get to this review in time. Looks good. Only thing I see is this PDF is a broken link. Looks like it's supposed to be generated as part of the build/deploy process.

Comment (d33bs) [reply README.md:8]: Thanks @vincerubinetti and no worries! This was a forward-facing content addition that relied on a successful deploy after merge (and wasn't available after initially merging as a result of the errors mentioned in #31). Now that the deploy worked, the link should now operate as expected.

**Pull request merged**